### DISCUSSION TITLE
Redirect log output to connected session.

### DISCUSSION THREAD

**Discussion by qazwsxedckll** [daniele77/cli#136] [Q&A] [answered]
I am integrating cli to my app. I have my own logging system and interface to choose where to output logs. My question is how to redirect logs to the connected session when using telnet. If I put logs to stdout, logs can only be seen using local session.

**Comment by daniele77** [comment]
Hi @qazwsxedckll ,
I'm glad you're using `cli` library and contributing to the community with your question.

For output on every open session, you can use the static member function: `Cli::cout()` as shown in the `complete` example that comes with the library ():

        rootMenu->Insert(
                "hello_everysession",
                [](std::ostream&){ Cli::cout() << "Hello, everybody" << std::endl; },
                "Print hello everybody on all open sessions" );

Being a static member, you have access to it anywhere in your code. Just call it on the same thread of the scheduler, to avoid race conditions.

Last but not least. If you like the library (and/or the support), please add a star to the project, and -- if you want -- you can consider sponsoring the project by clicking the heart button on top of the page. You can donate any amount, one-shot or periodic. This will help us keep on maintaining the library with the highest quality standard.

**Reply by qazwsxedckll** [reply]
But I want to output on the session where the command is sent. I have not read the code so I cannot give any advices. Thank you for your reply.

**Reply by daniele77** [reply]
The callback you provide for each command has got an `std::ostream&` parameter: it's the `ostream` associated with the session where the command is sent. So, for example:

    rootMenu->Insert(
            "hello_currentsession",
            [](std::ostream& o){ o << "Hello, everybody" << std::endl; },
            "Print hello everybody on the current session" );
    rootMenu->Insert(
            "hello_everysession",
            [](std::ostream&){ Cli::cout() << "Hello, everybody" << std::endl; },
            "Print hello everybody on all open sessions" );

**Comment by qazwsxedckll** [comment, marked as answer]
```c++
rootMenu->Insert('
"run", {"testcase"},
        [](std::ostream &out)
        {
            Logger::setOutput([&out](const char* msg, int len){ out << string(msg, len); });
            Logger::setFlush([&out](){ out.flush(); });
        },
```

This works for me, thanks for your reply
